### PR TITLE
Add track distance and distance from start of track to GPS information panel/toolbar

### PR DESCRIPTION
**Pull request by nyalldawson**
This PR adds the current total track length and distance from start of track to the GPS information panel, and also as a (not-on-by-default) option to show in the GPS toolbar information label.

Sponsored by NIWA

### CODE REVIEW

File: src/app/gps/qgsappgpsdigitizing.cpp [hunk 1]
```diff
@@ -97,6 +102,26 @@ QgsAppGpsDigitizing::~QgsAppGpsDigitizing()
   mRubberBand = nullptr;
 }
 
+double QgsAppGpsDigitizing::totalTrackLength() const
+{
+  QVector<QgsPointXY> points;
+  QgsGeometry::convertPointList( mCaptureListWgs84, points );
```
Comment (nirvn) [inline src/app/gps/qgsappgpsdigitizing.cpp:108]: Why discarding elevation when calculating distance here?

Comment (nyalldawson) [reply src/app/gps/qgsappgpsdigitizing.cpp:108]: We don't do ellipsoidal measurements with elevation (yet?)

Comment (nirvn) [reply src/app/gps/qgsappgpsdigitizing.cpp:108]: Might be worth noting somewhere that's it's a flattened length maths with a Todo: calculate 3d length whenever we can :)

**Review by nirvn — APPROVED** [review]

File: src/app/gps/qgsappgpsdigitizing.cpp (hunk 1 above)
Comment (nyalldawson) [reply src/app/gps/qgsappgpsdigitizing.cpp:108]: I'm just not sure if that's even possible -- I haven't found any libraries which offer 3d ellipsoidal calculations to date. 😱

**Comment by nyalldawson** [comment]
Gah, pressed the wrong button!

**Pull request merged**